### PR TITLE
[infra-proxy-service] Add GetCookbookFileContent gRPC & Http API

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?
- Add `GetCookbookFileContent` gPRC & HTTP API in order to fetch the file content of the cookbook.
- As some of the infra server cookbooks data required authentication to access the cookbook files content.

### :chains: Related Resources

#1544 

### :+1: Definition of Done
Able to access the cookbook file content using `GetCookbookFileContent` API.

### :athletic_shoe: How to Build and Test the Change
- `rebuild components/infra-proxy-service`
- `rebuild components/automate-gateway`

Test the changes:

- Add the chef-server & org credentials.
- Hit the GetCookbook API
- hit the GetCookbookFileContent API with URL params.

```
chef-automate dev grpcurl automate-gateway chef.automate.api.infra_proxy.InfraProxy.GetCookbookFileContent -- -d '{"server_id": "36b291ac-2352-43f1-b710-9abcb612e7cf", "org_id": "2ee2c6a5-5aad-4f59-95bc-5e8c0f4ff8c7", "name": "viv_aix", "version": "_latest", "url":"URL"}'
```
Response:
```
{
  "content": "name"
}

```

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: api/external/infra_proxy/request/cookbooks.proto [hunk 1]
```diff
@@ -20,3 +20,16 @@ message Cookbook {
     string name      = 3;
     string version   = 4;
 }
+
+message CookbookFileContent {
+    // Id of the org.
```
Comment (susanev) [inline api/external/infra_proxy/request/cookbooks.proto]:
```suggestion
    // ID of the org.
```

File: api/external/infra_proxy/request/cookbooks.proto [hunk 2]
```diff
@@ -20,3 +20,16 @@ message Cookbook {
...
+    string org_id = 1;
+    // Id of the server.
```
Comment (susanev) [inline api/external/infra_proxy/request/cookbooks.proto]:
```suggestion
    // ID of the server.
```

File: api/external/infra_proxy/request/cookbooks.proto [hunk 3]
```diff
@@ -20,3 +20,16 @@ message Cookbook {
...
+    string server_id = 2;
+    // Name of the cookbook.
+    string name = 3;
+    // Version of the cookbook.
+    string version = 4;
+    // Cookbook data file url.
```
Comment (susanev) [inline api/external/infra_proxy/request/cookbooks.proto]:
```suggestion
    // Cookbook data file URL.
```

File: api/interservice/infra_proxy/request/cookbooks.proto [hunk 4]
```diff
@@ -20,3 +20,16 @@ message Cookbook {
     string name = 3;
     string version = 4;
 }
+
+message CookbookFileContent {
+    // Id of the org.
```
Comment (susanev) [inline api/interservice/infra_proxy/request/cookbooks.proto]:
```suggestion
    // ID of the org.
```

File: api/interservice/infra_proxy/request/cookbooks.proto [hunk 5]
```diff
@@ -20,3 +20,16 @@ message Cookbook {
...
+    string org_id = 1;
+    // Id of the server.
```
Comment (susanev) [inline api/interservice/infra_proxy/request/cookbooks.proto]:
```suggestion
    // ID of the server.
```

File: api/interservice/infra_proxy/request/cookbooks.proto [hunk 6]
```diff
@@ -20,3 +20,16 @@ message Cookbook {
...
+    string server_id = 2;
+    // Name of the cookbook.
+    string name = 3;
+    // Version of the cookbook.
+    string version = 4;
+    // Cookbook data file url.
```
Comment (susanev) [inline api/interservice/infra_proxy/request/cookbooks.proto]:
```suggestion
    // Cookbook data file URL.
```

File: api/external/infra_proxy/request/cookbooks.proto (hunk 1 above)
Comment (vsingh-msys) [reply api/external/infra_proxy/request/cookbooks.proto]: Thanks for pointing it out :smiley:

**Pull request merged**